ext/json/ext/generator/generator.c

```diff
@@ -972,6 +973,8 @@ static inline VALUE vstate_get(struct generate_json_data *data)
     if (RB_UNLIKELY(!data->vstate)) {
         vstate_spill(data);
     }
+    GET_STATE(data->vstate);
+    state->depth = data->depth;
```

Comment:
Ouch actually this is bad, the State is mutated while it might be frozen, which also means that multiple Ractors could write to this concurrently. Fix coming up.